python/dgl/heterograph.py

```diff
@@ -6061,9 +6061,11 @@ def local_scope(self):
         old_eframes = self._edge_frames
         self._node_frames = [fr.clone() for fr in self._node_frames]
         self._edge_frames = [fr.clone() for fr in self._edge_frames]
-        yield
-        self._node_frames = old_nframes
-        self._edge_frames = old_eframes
+        try:
```

Comment:
could you add a dedicated testcase that this change tries to avoid?

Reply:
just alongside with https://github.com/dmlc/dgl/blob/9e26c8e38864d5625db59d5e5b22812b4d153533/tests/python/common/function/test_basics.py#L594

Reply:
Done. Test case added at line 680